src/main/java/net/glowstone/entity/passive/GlowVillager.java

```diff
@@ -99,4 +102,14 @@ protected Sound getDeathSound() {
     protected Sound getAmbientSound() {
         return Sound.ENTITY_VILLAGER_AMBIENT;
     }
+
+    @Override
+    public void damage(double amount, Entity source, DamageCause cause) {
+        if (!DamageCause.LIGHTNING.equals(cause)) {
+            super.damage(amount, source, cause);
```

Comment:
This would call the EntityDamageEvent which can be cancelled by plugins -- we should probably have an internal method that returns false when that event is cancelled, so subsequent actions (like converting) can be cancelled too.